tests/p2p/netfilter/test_match.py

```diff
@@ -191,3 +195,22 @@ def test_match_peer_id(self):
         # Fail because proto1 is connected to proto2, and the peer id cannot match.
         context = NetfilterContext(protocol=conn.proto1)
         self.assertFalse(matcher.match(context))
+
+
+class SyncV1NetfilterMatchTest(BaseNetfilterMatchTest):
```

Comment:
They are all the same across files. Can't we create a method that receives a `TestCase` and return these three variations registered as valid tests?

Another way to go would be to create a `__sync_variations__` and make the scanner find them and generate all three variations automatically. What do you think?

Reply:
I think there might be a lot of ways go about this. I just didn't want to spend too much time on this.

A decorator wouldn't work unless it modifies the methods (which I'm not a big fan of, because it has to duplicate the logic of detecting which methods are tests and which are not, and then decorate them individually).

I think using something outside of the tests would be the best solution (like running all the tests with different configs), and some tests will be skipped when sync-v1 is enabled and others when sync-v2 is enabled. But this looks like it would be a lot more work and would run multiple times tests that don't have anything to do with sync.

Anyway your suggestion aren't too clear in my mind. We could have a talk to see if I understand or if you have any example we could discuss on top of them.

Reply:
I think this is fixed now.